kanidmd/src/lib/idm/authsession.rs

```diff
@@ -1077,6 +1177,74 @@ mod tests {
         audit.write_log();
     }
 
+    #[test]
+    fn test_idm_authsession_totp_password_badlist() {
```

Comment:
minor nitpick, I think this should be password_mfa_badlist, not totp_password_badlist.